### PR TITLE
CLN: tighten Exception catching in indexing.py

### PR DESCRIPTION
**Pull request by jbrockmendel**
get rid of is_lexsorted_for_tuple while we're at it

### CODE REVIEW

**Comment by gfyoung** [comment]
Are exceptions simply being handled more specifically now?

Or are they be handled higher up in the stack now?

File: doc/source/reference/indexing.rst [hunk 1]
```diff
@@ -93,7 +93,6 @@ Compatibility with MultiIndex
    :toctree: api/
 
    Index.set_names
-   Index.is_lexsorted_for_tuple
```
Comment (jorisvandenbossche) [inline doc/source/reference/indexing.rst]: As mentioned a few lines above, this is apparently for compatibility with MultiIndex. That is no longer needed?

**Comment by jorisvandenbossche** [comment]
Echoing @gfyoung's question, it would be helpful if you add some explanation why you are removing / changing things in your PRs.

**Comment by jbrockmendel** [comment]
> Are exceptions simply being handled more specifically now?  Or are they be handled higher up in the stack now?

They are no longer reached.  AFAICT this is leftover from Panel indexing

File: doc/source/reference/indexing.rst (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/reference/indexing.rst]: It is not needed by pandas maybe, but it is part of the public API (to make the API of Index and MultiIndex more alike)

Comment (jreback) [reply doc/source/reference/indexing.rst]: I don't think its a big deal to remove this method. Could have a note about it I guess.

Comment (jbrockmendel) [reply doc/source/reference/indexing.rst]: > Should we then remove it from MultiIndex as well?

It is not overridden by MultiIndex

Comment (jorisvandenbossche) [reply doc/source/reference/indexing.rst]: > It is not overridden by MultiIndex

That's not my point. What I am trying to say is: in the past, we tried to keep a consistent api between Index / MultiIndex. So why not do that here?(either keep both, or remove both)

Comment (jbrockmendel) [reply doc/source/reference/indexing.rst]: I'm definitely confused.  If this PR is merged, then there will be no `is_lexsorted_for_tuple` on _any_ class.  Which should classify as "remove both", shouldn't it?

**Comment by jreback** [comment]
can you add a whatsnew note, technically this *was* public api; ping on green.

**Comment by jbrockmendel** [comment]
ping

File: doc/source/reference/indexing.rst (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/reference/indexing.rst]: Sorry, I mis-interpreted your "this is not overriden". I was only thinking about the implications for the user (the fact that it is overriden or not does not necessarily mean this method was not there for API compatibility), but of course that means you also removed the method for MultiIndex. So fine with this!

**Pull request merged**